exp-models/addon/utils/csv-writer.js

```diff
@@ -0,0 +1,107 @@
+import Ember from 'ember';
+
+/**
+ * New line string
+ *
+ * @type {string}
+ */
+export const NL = '\r\n';
+
+/**
+ * Flatten a nested object into a single level, with dotted paths for keys
+ *
+ * @param obj {!Object} - The object to flatten
+ * @param prefix {string} - The prefix for the keys
+ * @returns {Object} - The flattened object
+ */
+function squashPrefix(obj, prefix) {
+    let ret = {};
+
+    for (const key in obj) {
+        if (!obj.hasOwnProperty(key)) {
+            continue;
+        }
+
+        const newPrefix = prefix ? `${prefix}.${key}` : key;
+
+        let value = Ember.get(obj, key);
+
+        if (value && value.toJSON) {
+            value = value.toJSON();
+        }
+
+        if (Ember.$.isPlainObject(value)) {
+            Object.assign(ret, squashPrefix(value, newPrefix));
+        } else {
+            ret[newPrefix] = value;
+        }
+    }
+
+    return ret;
+}
+
+/**
+ * Flatten a nested object into a single level, with dotted paths for keys
+ *
+ * @param obj {!Object} - The object to flatten
+ * @returns {Object} - The flattened object
+ */
+export function squash(obj) {
+    return squashPrefix(obj);
+}
+
+/**
+ * Extracts the unique object keys from an array of objects
+ *
+ * @param dataArray {Array<Object>} - An array of objects
+ * @returns {Array<string>} - The unique keys
+ */
+export function uniqueFields(dataArray) {
```

Comment:
One could simply not use this if the fields are already well defined--which is why I separated it out into it's own function. If you just want to parse the first row,

```js
[dataArray[0]]
```

could be passed to it